### PR TITLE
Update filter for Kotlin suspending functions

### PR DESCRIPTION
**Pull request by Godin**
`Example.kt`

```
import kotlinx.coroutines.delay

fun nop() {
}

suspend fun example() { // line 6
  delay(0) // line 7
  nop()
}
```

`javap -v -p ExampleKt.class`

```
  public static final java.lang.Object example(kotlin.coroutines.Continuation<? super kotlin.Unit>);
    descriptor: (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
    flags: ACC_PUBLIC, ACC_STATIC, ACC_FINAL
    Code:
      stack=5, locals=4, args_size=1
         0: aload_0
         1: instanceof    #12                 // class ExampleKt$example$1
         4: ifeq          36
         7: aload_0
         8: checkcast     #12                 // class ExampleKt$example$1
        11: astore_2
        12: aload_2
        13: getfield      #16                 // Field ExampleKt$example$1.label:I
        16: ldc           #17                 // int -2147483648
        18: iand
        19: ifeq          36
        22: aload_2
        23: dup
        24: getfield      #16                 // Field ExampleKt$example$1.label:I
        27: ldc           #17                 // int -2147483648
        29: isub
        30: putfield      #16                 // Field ExampleKt$example$1.label:I
        33: goto          45
        36: new           #12                 // class ExampleKt$example$1
        39: dup
        40: aload_0
        41: invokespecial #21                 // Method ExampleKt$example$1."<init>":(Lkotlin/coroutines/Continuation;)V
        44: astore_2
        45: aload_2
        46: getfield      #25                 // Field ExampleKt$example$1.result:Ljava/lang/Object;
        49: astore_1
        50: invokestatic  #31                 // Method kotlin/coroutines/intrinsics/IntrinsicsKt.getCOROUTINE_SUSPENDED:()Ljava/lang/Object;
        53: astore_3
        54: aload_2
        55: getfield      #16                 // Field ExampleKt$example$1.label:I
        58: tableswitch   { // 0 to 1
                       0: 80
                       1: 113
                 default: 138
            }
        80: aload_1
        81: dup
        82: instanceof    #33                 // class kotlin/Result$Failure
        85: ifeq          95
        88: checkcast     #33                 // class kotlin/Result$Failure
        91: getfield      #37                 // Field kotlin/Result$Failure.exception:Ljava/lang/Throwable;
        94: athrow
        95: pop
        96: lconst_0
        97: aload_2
        98: aload_2
        99: iconst_1
       100: putfield      #16                 // Field ExampleKt$example$1.label:I
       103: invokestatic  #43                 // Method kotlinx/coroutines/DelayKt.delay:(JLkotlin/coroutines/Continuation;)Ljava/lang/Object;
       106: dup
       107: aload_3
       108: if_acmpne     130
       111: aload_3
       112: areturn
       113: aload_1
       114: dup
       115: instanceof    #33                 // class kotlin/Result$Failure
       118: ifeq          128
       121: checkcast     #33                 // class kotlin/Result$Failure
       124: getfield      #37                 // Field kotlin/Result$Failure.exception:Ljava/lang/Throwable;
       127: athrow
       128: pop
       129: aload_1
       130: pop
       131: invokestatic  #45                 // Method nop:()V
       134: getstatic     #51                 // Field kotlin/Unit.INSTANCE:Lkotlin/Unit;
       137: areturn
       138: new           #53                 // class java/lang/IllegalStateException
       141: dup
       142: ldc           #55                 // String call to 'resume' before 'invoke' with coroutine
       144: invokespecial #58                 // Method java/lang/IllegalStateException."<init>":(Ljava/lang/String;)V
       147: athrow
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           45      93     2 $continuation   Lkotlin/coroutines/Continuation;
      LineNumberTable:
        line 6: 53
        line 7: 96
        line 6: 111
        line 8: 130
        line 9: 134
```

### CODE REVIEW

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java [hunk 1]
```diff
@@ -137,4 +136,155 @@ public void should_filter() {
...
+		m.visitJumpInsn(Opcodes.IFEQ, createStateInstance);
+
+		m.visitVarInsn(Opcodes.ALOAD, continuationArgumentIndex);
+		m.visitTypeInsn(Opcodes.CHECKCAST, "ExampleKt$example$1");
+		m.visitVarInsn(Opcodes.ASTORE, continuationIndex);
+
+		m.visitVarInsn(Opcodes.ALOAD, continuationIndex);
+		m.visitFieldInsn(Opcodes.GETFIELD, "ExampleKt$example$1", "label", "I");
+
+		m.visitLdcInsn(1 << 31);
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java]: Gives an outoboxing warning.

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java [hunk 2]
```diff
@@ -137,4 +136,155 @@ public void should_filter() {
...
+		m.visitInsn(Opcodes.IAND);
+		m.visitJumpInsn(Opcodes.IFEQ, createStateInstance);
+
+		m.visitVarInsn(Opcodes.ALOAD, continuationIndex);
+		m.visitInsn(Opcodes.DUP);
+		m.visitFieldInsn(Opcodes.GETFIELD, "ExampleKt$example$1", "label", "I");
+
+		m.visitLdcInsn(1 << 31);
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java]: Another outoboxing warning here.

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java (hunk 1 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java]: Fixed.

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java (hunk 2 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java]: Fixed.

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java [hunk 3]
```diff
@@ -170,15 +170,15 @@ public void should_filter_suspending_functions() {
 		m.visitVarInsn(Opcodes.ALOAD, continuationIndex);
 		m.visitFieldInsn(Opcodes.GETFIELD, "ExampleKt$example$1", "label", "I");
 
-		m.visitLdcInsn(1 << 31);
+		m.visitLdcInsn(Integer.MIN_VALUE);
```
[the remark this answers is not in the document]
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilterTest.java]: Doh! now should be really fixed.

**Pull request merged**